src/lib/components/chat/ChatMessage.svelte

```diff
@@ -410,29 +410,25 @@
 			{/if}
 
 			<div class="flex w-full flex-row flex-nowrap">
-				{#if !editMode}
-					<p
-						class="disabled w-full appearance-none whitespace-break-spaces text-wrap break-words bg-inherit px-5 py-3.5 text-gray-500 dark:text-gray-400"
-					>
-						{message.content.trim()}
-					</p>
-				{:else}
-					<form
-						class="flex w-full flex-col"
-						bind:this={editFormEl}
-						on:submit|preventDefault={() => {
-							dispatch("retry", { content: editContentEl.value, id: message.id });
-							$convTreeStore.editing = null;
-						}}
+				<form
+					class="flex w-full flex-col"
+					bind:this={editFormEl}
+					on:submit|preventDefault={() => {
+						dispatch("retry", { content: editContentEl.innerText, id: message.id });
+						$convTreeStore.editing = null;
+					}}
+				>
+					<span
+						role="textbox"
+						tabindex={1}
+						class="w-full whitespace-break-spaces text-wrap break-words break-words rounded-xl px-5 py-3.5 text-gray-500 outline-none *:h-max dark:text-gray-400"
+						class:bg-gray-100={editMode}
+						class:dark:bg-gray-800={editMode}
+						bind:this={editContentEl}
+						on:keydown={handleKeyDown}
+						contenteditable={editMode}>{message.content.trim()}</span
```

Comment:
I can paste html and it renders:

<img width="1580" alt="image" src="https://github.com/huggingface/chat-ui/assets/3841370/18f75a70-fa62-4e50-af5f-3c69ba8a00c0">


I would stay with a`<textarea>` and do the elastic height like we already do on https://github.com/huggingface/chat-ui/blob/e9c846acd33f806b30498f67140bac9579c1c33b/src/lib/components/chat/ChatInput.svelte#L46